federation/pkg/dnsprovider/plugins.go

```diff
@@ -29,7 +29,7 @@ import (
 // The config parameter provides an io.Reader handler to the factory in
 // order to load specific configurations. If no configuration is provided
 // the parameter is nil.
-type Factory func(config io.Reader) (Interface, error)
+type Factory func(config io.Reader, zoneName string) (Interface, error)
```

Comment:
What is you motivation for restricting a dnsProvider to a single zone name?
I think I'd prefer not to have a zoneName associated with a DnsProvider.  The current implementation allows a single DnsProvider to manage multiple zoneNames, which is, I think, the correct general model.